docs/IE_PLUGIN_DG/NewTransformation.md

```diff
@@ -1,15 +1,559 @@
+In this case `std::shared_ptr<Node>` will be automatically converted to `Output<Node>` if node has exactly one output port otherwise conversion will raise an exception.   
+
+`2.` ngraph::Node replacement
+
+nGraph provides two ways for node replacement: via ngraph helper function and directly via port methods. We are going to review both of them.
+
+Lets start with nGraph helper functions. The most popular function is `ngraph::replace_node(old_node, new_node)`.
+
+Usage example:
+~~~~~~~~~~~~~{.cpp}
+auto neg = std::dynamic_pointer_cast<ngraph::opset1::Negative> (m.get_match_root());
+if (!neg) {
+    return false;
+}
+
+auto mul = std::make_shared<ngraph::opset1::Multiply>(neg->input_value(0),
+                                                      opset1::Constant::create(neg->get_element_type(), Shape{1}, {-1}));
+mul->set_friendly_name(neg->get_friendly_name());
+ngraph::copy_runtime_info(neg, mul);
+// Replaces Negative operation with Multiply operation
+ngraph::replace_node(neg, mul);
+~~~~~~~~~~~~~ 
+`ngraph::replace_node` has a constraint that number of output ports for both of ops must be the same otherwise it will raise an exception.
+
+
+The alternative way to do the same replacement is next:
+~~~~~~~~~~~~~{.cpp}
+// All neg->output(0) consumers will be moved to mul->output(0) port
+neg->output(0).replace(mul->output(0));
+~~~~~~~~~~~~~
+
+Another transformation example is insertion.
+~~~~~~~~~~~~~{.cpp}
+// Lets suppose that we have a node with single output port and we want to insert additional operation new_node after it
+void insert_example(std::shared_ptr<ngraph::Node> node) {
+    // Get all consumers for node
+    auto consumers = node->output(0).get_target_inputs();
+    // Create new node. Let it be opset1::Relu.
+    auto new_node = std::make_shared<ngraph::opset1::Relu>(node);
+    // Reconnect all consumers to new_node
+    for (auto input : consumers) {
+        input.replace_source_output(new_node);
+    }
+}
+~~~~~~~~~~~~~ 
+
+The alternative way to insert operation is to make a node copy and use `replace_node`:
+~~~~~~~~~~~~~{.cpp} 
+void insert_example(std::shared_ptr<ngraph::Node> node) {
+    // Make a node copy 
+    auto node_copy = node->clone_with_new_inputs(node->input_values());
+    // Create new node
+    auto new_node = std::make_shared<ngraph::opset1::Relu>(node_copy);
+    ngraph::replace_node(node, new_node);
+}
+~~~~~~~~~~~~~
+
+`3.` ngraph::Node elimination
+
+Another type of node replacement is its elimination.
+
+To eliminate operation nGraph has special method that consider all limitations related to InferenceEngine.
+~~~~~~~~~~~~~{.cpp}
+// Suppose we have a node that we want to remove
+bool success = replace_output_update_name(node->output(0), node->input_value(0));
+~~~~~~~~~~~~~ 
+
+`replace_output_update_name` in case of successful replacement it automatically preserves friendly name and runtime info.
```

Comment:
`in case of successful replacement it automatically preserves` -> `in case of successful replacement automatically preserves`

Reply:
I agree with yours but my sound more natural for me :)